packages/notifications/src/notifications.spec.ts

```diff
@@ -0,0 +1,57 @@
+import { createStore, applyMiddleware } from 'redux';
+import { TNotificationAction, TNotificationState } from './types';
+import { addNotification } from './action-creators';
+import reducer from './reducer';
+import middleware from './middleware';
+
+type FooNotification = {
+  id: number;
+  foo: string;
+};
+
+describe('dispatching add/remove notification actions', () => {
+  it('should add a notification and wait for it to be removed after 100ms', () => {
+    jest.useFakeTimers();
+    const notification: FooNotification = { id: 0, foo: 'bar' };
+
+    const store = createStore<
+      TNotificationState<FooNotification>,
+      TNotificationAction<FooNotification>,
+      unknown,
+      unknown
+    >(reducer, undefined, applyMiddleware(middleware));
+    expect(store.getState()).toEqual([]);
+    const action = addNotification(notification, {
+      dismissAfter: 100,
+    });
+    store.dispatch(action);
+    expect(store.getState()).toEqual(
+      expect.arrayContaining([expect.objectContaining({ foo: 'bar' })])
+    );
+    jest.advanceTimersByTime(101);
+    expect(store.getState()).toEqual([]);
+  });
+
+  it('should add a notification and remove it after calling onDismiss', () => {
+    const notification: FooNotification = { id: 0, foo: 'bar' };
+
+    const store = createStore<
+      TNotificationState<FooNotification>,
+      TNotificationAction<FooNotification>,
+      unknown,
+      unknown
```

Comment:
I read what the unknown type is can you share why we use it here?

Reply:
Yes, it's because the `createStore` signature expects 4 generic types:

```ts
<S, A extends Action, Ext, StateExt>
```

But I'm only interested in the first 2. The `unknown` is kind of like a "safer" `any`.

<img width="441" alt="image" src="https://user-images.githubusercontent.com/1110551/61723402-5f646900-ad6c-11e9-920e-2ba0d1072831.png">